tiledb/sm/compressors/zstd_compressor.cc

```diff
@@ -80,22 +80,26 @@ Status ZStd::compress(
 }
 
 Status ZStd::decompress(
-    ConstBuffer* input_buffer, PreallocatedBuffer* output_buffer) {
+    tdb_shared_ptr<ResourcePool<ZStd::ZSTD_Decompress_Context>>
```

Comment:
I know I said I had no reason to use `std::shared_ptr,` but I actually had and forgot about it, sorry for that.
The reason is the recommendation to use `std::shared_ptr` instead of `tdb_shared_ptr` in new code after the new allocator has been introduced: https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/common/dynamic_memory/dynamic_memory.h#L259
So I would revert these to the dev versions.

Reply:
We should not use `std::shared_ptr` but instead `tdb::shared_ptr` (or `tiledb::common:shared_ptr`). It's true that it's an alias to `std::shared_ptr` but by avoiding the `std` version this is how we ensure all usages are going through our allocator.

Here the branch `release-2.6` does not have the allocator so we use the older heap profiler version.

Reply:
The choice of alias is only a code style issue; it doesn't ensure anything about the origin of the pointer. Both `std::make_shared` and `tdb::make_shared` return the same type, and either return value can be assigned to any alias of `shared_ptr`.

The choice of style we've picked, but not moved to, is to put `shared_ptr` into the global namespace. It's universally used, there's no confusion around it, and we can avoid excess syntax. The declaration to do so is in `common.h`, which we haven't started using everywhere yet.

Reply:
@eric-hughes-tiledb even if I include `common.h`, I get a compilation error when using `shared_ptr` in the global namespace. Is it maybe missing from `common-std.h` or am I doing something wrong?